Laser/Energy Buffs

KillerTheFlareon opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
Reworks laser/energy gun capacity, and in the case of destroy mode, damage.

## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
In sci-fi the main advantage of lasers over bullets is
- No recoil
- Can hold a lot of ammo in a more compact space
- Easy to use
- Portable

Our lasers, right now have none of those advantages, minus recoil.
Kinetics outclass them in every way possible, from ease to use, to space, to even damage output

So the idea behind the buff is to make lasers stronger in their niche
That means taking what the average secoff carries on red, and putting it in the gun.

_The poll I made was a tie between 2-4 with a bunch of votes in 3, which means the average secoff carries 3+1 magazines (due to the one in the gun, and yes my poll specified that it was EXTRA mags, not including the one your gun comes with)_
Edit: I changed this to 2 magazines and made the gun 2x4, to make carrying lasers less cumbersome at pig's suggestion. It should overall lead to lasers being used more often, or at the least being carried as a backup weapon anyway.

To make lasers attractive, I put the average amount of ammo the secoff carries into one gun, so you do not have to reload and it takes less overall space. A kinetic will still be BETTER than a laser rifle, but it is now a valid option amongst kinetics.

I then found out the dominator, which can now go through glass and is hitscan, does 20 heat damage- basically making it a pocket laser rifle. Obviously the actual RIFLE shouldn't do less damage than the sidearm in your pocket, especially considering they're both lasers on the same UV wavelength with the same power output. 

20 heat sounds like a lot, but it's reduced by any sort of armor, unlike kinetics which have innate AP.

I also gave buffs to the other laser weapons, such as x-ray/cannon being 25 (a full lector magazine) and the immolator (literally lasrifle+) the same 50 capacity treatment, although it trades some damage output (15 instead of 20) for the ability to create fire stacks on demand.

"Energy Gun" was just a worse dominator. I changed that to be.. a less worse dominator. It's a sidearm, WCYD.
Eshotty capacity was also buffed, to compensate for the lack of self charge, as the whole reason it had a low capacity was BECAUSE it could self charge.

I also gave destroy mode 60 heat instead of 50, as the entire point of destroy mode was to be a 2-shot crit. However, any sort of armor (even 1%) would make it a 3-shot crit, which doesn't work when the gun that uses destroy mode has 2-shots. 60+60 = 120, which should give destroy mode utility once more as it'll go through weak armor.
## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: Killer Tamashi
- tweak: Changed All laser sizes to 2x4 instead of 4x4.
- tweak: Changed Laser capacity to 50 (2x Magazines)
- tweak: Changed Laser damage to 14 -> 20 Heat (On par with Dominator)
- tweak: Changed Immolator capacity to 50 (2x Magazines)
- tweak: Changed Immolator damage to 14 -> 15 Heat 
- tweak: Changed Laser Cannon capacity to 25 (1x Magazine)
- tweak: Changed X-Ray Laser capacity to 25 (1x Magazine)
- tweak: Changed Energy Gun capacity to 20/15 (On par with Dominator)
- tweak: Changed Energy Shotgun capacity to 15/20 (Narrow / Disabler for clarity)
- tweak: Changed Destroy Mode Damage 50 -> 60 (Should allow for 2-Hit Crits, as intended)

KillerTheFlareon committed:
Laser Buffs

Laser is shit in it's current state, this makes laser not shit. Simple as.

KillerTheFlareon committed:
awa

a

RHSvenson commented:
If we are buffing the base laser beam, there is no reason to have a Starlight exclusive beam. After they were changed to beams, they are now functionally identical, and the Starlight-exclusive beam (RedLaserBeam) was retained solely for the sake of maintaining the status-quo balance of the X-01 and Pulse weapons.

If we are buffing the native leaser beams to 20, then just delete the Starlight beam and make our weapons use the Wizden file beam, which now will also do 20 damage. Or vice-versa, of course. Either way, if they are the same, we should only be using one.

Not a critique, just a note for a future yaml simplification.

RHSvenson commented:
Also possibly conflicts with https://github.com/ss14Starlight/space-station-14/pull/984#issue-3209177264 ?

KillerTheFlareon commented:
> Also possibly conflicts with [#984 (comment)](https://github.com/ss14Starlight/space-station-14/pull/984#issue-3209177264) ?

Yeaahhh pig was faster to the draw than I was. Up to the maintainers to choose which one they want to merge.

KillerTheFlareon commented:
> If we are buffing the base laser beam, there is no reason to have a Starlight exclusive beam. After they were changed to beams, they are now functionally identical, and the Starlight-exclusive beam (RedLaserBeam) was retained solely for the sake of maintaining the status-quo balance of the X-01 and Pulse weapons.
> 
> If we are buffing the native leaser beams to 20, then just delete the Starlight beam and make our weapons use the Wizden file beam, which now will also do 20 damage. Or vice-versa, of course. Either way, if they are the same, we should only be using one.
> 
> Not a critique, just a note for a future yaml simplification.

I'd imagine it's the sprite that people want? If a maintainer says "might as well just use the wizden laser now" then I'll just do that. But iirc it's just to have it be visually distinct

Lasers fire BEAMS and energy weapons fire.. uh.. those laser bullet things.

RHSvenson commented:
> > If we are buffing the base laser beam, there is no reason to have a Starlight exclusive beam. After they were changed to beams, they are now functionally identical, and the Starlight-exclusive beam (RedLaserBeam) was retained solely for the sake of maintaining the status-quo balance of the X-01 and Pulse weapons.
> > If we are buffing the native leaser beams to 20, then just delete the Starlight beam and make our weapons use the Wizden file beam, which now will also do 20 damage. Or vice-versa, of course. Either way, if they are the same, we should only be using one.
> > Not a critique, just a note for a future yaml simplification.
> 
> I'd imagine it's the sprite that people want? If a maintainer says "might as well just use the wizden laser now" then I'll just do that. But iirc it's just to have it be visually distinct
> 
> Lasers fire BEAMS and energy weapons fire.. uh.. those laser bullet things.

Both laser beams are identical graphically right now, it is basically only the damage. The disabler stuff is still very different, due to projectile vs. hitscan, but both wizden and Starlight laser beams are hitscan beams.

KillerTheFlareon commented:
> > > If we are buffing the base laser beam, there is no reason to have a Starlight exclusive beam. After they were changed to beams, they are now functionally identical, and the Starlight-exclusive beam (RedLaserBeam) was retained solely for the sake of maintaining the status-quo balance of the X-01 and Pulse weapons.
> > > If we are buffing the native leaser beams to 20, then just delete the Starlight beam and make our weapons use the Wizden file beam, which now will also do 20 damage. Or vice-versa, of course. Either way, if they are the same, we should only be using one.
> > > Not a critique, just a note for a future yaml simplification.
> > 
> > 
> > I'd imagine it's the sprite that people want? If a maintainer says "might as well just use the wizden laser now" then I'll just do that. But iirc it's just to have it be visually distinct
> > Lasers fire BEAMS and energy weapons fire.. uh.. those laser bullet things.
> 
> Both laser beams are identical graphically right now, it is basically only the damage. The disabler stuff is still very different, due to projectile vs. hitscan, but both wizden and Starlight laser beams are hitscan beams.

oh. Uhh. Huh. Alright

PubliclyExecutedPig commented:
what order is the capacity buffs on the eshotgun? also eshotgun looks a lil overtuned, i would say like 16 disabler bolts, 12 wide shots and 10 narrows

KillerTheFlareon commented:
> what order is the capacity buffs on the eshotgun? also eshotgun looks a lil overtuned, i would say like 16 disabler bolts, 12 wide shots and 10 narrows

Eshotty disabler was given normal disabler capacity. If you fully hit someone with all disabler shots it does the equivalent stamina damage of 1 disabler shot from a disabler

As for capacity uh. 20 is disabler, don't remember the order of the other two I'll need to check ngl

Littlemen07 commented:
I do believe it would be more healthy if there was more narrow shots than there is wide.
the problem with the eshotgun is its one tap potential, having insane dps with its wide settign at very close range.

So i think the wide setting should have less ammo and the narrow become its main fighting ability, it will still make it an excellent weapons without the 300 heat corpse meme.

And im still an advocate for self recharge, the recharge on the charger is very slow and there isnt many secured charger where i can confidently leave it for a minute and be sure it wont get taken by a tider that was in brig for some reason.

KillerTheFlareon commented:
> I do believe it would be more healthy if there was more narrow shots than there is wide. the problem with the eshotgun is its one tap potential, having insane dps with its wide settign at very close range.
> 
> So i think the wide setting should have less ammo and the narrow become its main fighting ability, it will still make it an excellent weapons without the 300 heat corpse meme.
> 
> And im still an advocate for self recharge, the recharge on the charger is very slow and there isnt many secured charger where i can confidently leave it for a minute and be sure it wont get taken by a tider that was in brig for some reason.

I did always find it odd narrow was the mode that costed more than wide. You'd think it would be the opposite..

PubliclyExecutedPig commented:
> > I do believe it would be more healthy if there was more narrow shots than there is wide. the problem with the eshotgun is its one tap potential, having insane dps with its wide settign at very close range.
> > So i think the wide setting should have less ammo and the narrow become its main fighting ability, it will still make it an excellent weapons without the 300 heat corpse meme.
> > And im still an advocate for self recharge, the recharge on the charger is very slow and there isnt many secured charger where i can confidently leave it for a minute and be sure it wont get taken by a tider that was in brig for some reason.
> 
> I did always find it odd narrow was the mode that costed more than wide. You'd think it would be the opposite..

I mean narrow still 2 shots, right? I think the idea of it still doing massive damage whilst not risking yourself as much is why it costs more

JustAdler commented:
<img width="546" height="272" alt="image" src="https://github.com/user-attachments/assets/bafe5a91-2aa0-4164-bcf0-b0f6f14a9989" />
<img width="563" height="291" alt="image" src="https://github.com/user-attachments/assets/c288e024-eb42-4474-bfa1-ae6824a1872d" />

Narrow is more expensive because of it's pretty damn good accuracy still

Littlemen07 commented:
> <img alt="image" width="546" height="272" src="https://private-user-images.githubusercontent.com/55048765/468556453-bafe5a91-2aa0-4164-bcf0-b0f6f14a9989.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMwOTI1NzIsIm5iZiI6MTc1MzA5MjI3MiwicGF0aCI6Ii81NTA0ODc2NS80Njg1NTY0NTMtYmFmZTVhOTEtMmFhMC00MTY0LWJjZjAtYjBmNmYxNGE5OTg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIxVDEwMDQzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2NTI2Mjk0ZmUzYTA1NjgxNDVhYjc0MDhjZjY1YTczOWM4MTExYTIxZmNjMWY4MjE1NGViZmUyZTJjOTk2M2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0RJpSyF9Uldxhn5BqXc-YJEx_FexzNbJcAwph_J38BE"> <img alt="image" width="563" height="291" src="https://private-user-images.githubusercontent.com/55048765/468556586-c288e024-eb42-4474-bfa1-ae6824a1872d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMwOTI1NzIsIm5iZiI6MTc1MzA5MjI3MiwicGF0aCI6Ii81NTA0ODc2NS80Njg1NTY1ODYtYzI4OGUwMjQtZWI0Mi00NDc0LWJmYTEtYWU2ODI0YTE4NzJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIxVDEwMDQzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiNmM4NDg3NWFiNzk0ZWM0OGI5MWZkMjY5ZGFkMWY4NGRhZDRmMjIxOTljMWY0MzcwNmEyMjExMjlmNjI4MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NSaBAVFhsxG9qYTMzYRQ3agpVamqMfBpBSbteSf7iLY">
> 
> Narrow is more expensive because of it's pretty damn good accuracy still

Would it be possible to change the wide setting to do structural damage only?
And maybe the narrow setting to be 3 lasers instead of the 4?

KillerTheFlareon commented:
> <img alt="image" width="546" height="272" src="https://private-user-images.githubusercontent.com/55048765/468556453-bafe5a91-2aa0-4164-bcf0-b0f6f14a9989.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxMjk5NzEsIm5iZiI6MTc1MzEyOTY3MSwicGF0aCI6Ii81NTA0ODc2NS80Njg1NTY0NTMtYmFmZTVhOTEtMmFhMC00MTY0LWJjZjAtYjBmNmYxNGE5OTg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIxVDIwMjc1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5OGY1ZWIyMzRhNDY0YzE2YTM0NTQ0NWJhNWNkNTQyYjkwYjY3MjZmMjVjMWQ3YWNkY2JlYmE2MTEzNjM1MjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NG3lOhnZKvrSlpAO-qsf06g1b7vWixTjTATXArRiTn4"> <img alt="image" width="563" height="291" src="https://private-user-images.githubusercontent.com/55048765/468556586-c288e024-eb42-4474-bfa1-ae6824a1872d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTMxMjk5NzEsIm5iZiI6MTc1MzEyOTY3MSwicGF0aCI6Ii81NTA0ODc2NS80Njg1NTY1ODYtYzI4OGUwMjQtZWI0Mi00NDc0LWJmYTEtYWU2ODI0YTE4NzJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzIxVDIwMjc1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYyMDQwODBjYzA0MWQ3MTQ1OTRiZTc5YjFkMzhmMjQ0Yzg0OTA3ZGYyYWZlMjA3YzdkNzFmMmRhMDU2YjMwNDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Oyl7IVQmm5tTUaAerq9rTkNYxF6_j00hH3EKXZyRaSE">
> Narrow is more expensive because of it's pretty damn good accuracy still

Okay.. so what am I doing here? Leaving as is or..

KillerTheFlareon committed:
Changes

Talked it over with pig, liked his idea. Combined our PRs together to prevent conflicts.

KillerTheFlareon committed:
Merge branch 'Starlight' into Laser-buffs

KillerTheFlareon committed:
Merge branch 'Starlight' into Laser-buffs

KillerTheFlareon committed:
changed costs a lil

KillerTheFlareon commented:
updated to dev branch instead of main

PubliclyExecutedPig commented:
killer ik we already kinda agreed on something for the eshotgun, but now that its wardens gun, well you'll need to redo it so that wide spread isnt accidentally re-added, but that also means you can tone it down since its not hos's anymore

KillerTheFlareon commented:
> killer ik we already kinda agreed on something for the eshotgun, but now that its wardens gun, well you'll need to redo it so that wide spread isnt accidentally re-added, but that also means you can tone it down since its not hos's anymore

so THAT'S WHY I GOT A CONFLICT. Hmm. Suppose double capacity works in that case.

PubliclyExecutedPig commented:
> > killer ik we already kinda agreed on something for the eshotgun, but now that its wardens gun, well you'll need to redo it so that wide spread isnt accidentally re-added, but that also means you can tone it down since its not hos's anymore
> 
> so THAT'S WHY I GOT A CONFLICT. Hmm. Suppose double capacity works in that case.

also make sure the wide spread is hashed out or straight up removed, do NOT re-add it since yknow... wardens gun

KillerTheFlareon commented:
> > > killer ik we already kinda agreed on something for the eshotgun, but now that its wardens gun, well you'll need to redo it so that wide spread isnt accidentally re-added, but that also means you can tone it down since its not hos's anymore
> > 
> > 
> > so THAT'S WHY I GOT A CONFLICT. Hmm. Suppose double capacity works in that case.
> 
> also make sure the wide spread is hashed out or straight up removed, do NOT re-add it since yknow... wardens gun

heard you the first time lol, i'll fix it

KillerTheFlareon committed:
a

a

ShadesMars commented on Resources/Changelog/ChangelogStarlight.yml:
This is not suposed to be here.

KillerTheFlareon replied:
whaddya mean? it's the changelog, my PR didn't touch that

ShadesMars reviewed (changes requested):
All code looks good but balance wise i am unsure so imma have to talk with some peoples before approving.

KillerTheFlareon committed:
Update ChangelogStarlight.yml

KillerTheFlareon commented:
what- oh

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Laser-buffs

KillerTheFlareon commented:
fixed, i think
